### PR TITLE
WIP: Fix Mike Rebase

### LINKED ISSUE #60: 🐛💻 – mike deploy fails because the remote contains work that you do not have locally
**Describe the problem**

<https://github.com/TIDES-transit/TIDES/actions/runs/3129865893/jobs/5079466105#step:8:25>

```
error: failed to push branch gh-pages to origin: "To https://github.com/TIDES-transit/TIDES
 ! [rejected]        gh-pages -> gh-pages (fetch first)
error: failed to push some refs to 'https://github.com/TIDES-transit/TIDES'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details."
```

I thought the `--rebase` flag was supposed to handle that. Without any versioning, we just keep overwriting `main`, so maybe we want `--ignore` instead of `--rebase`?

The `mike` docs also suggest you might need to manually fetch before running:

```sh
git fetch origin gh-pages --depth=1
```

Maybe we can just change the `fetch-depth` parameter in the `checkout` action to `1`? But I guess that would fetch main, not gh-pages. So probably need to add the `git fetch` command above to the `Build docs` step.

**Steps to reproduce**

### PR DESCRIPTION
**Pull request by e-lo**
This PR resolves the error in #60 where Mike doesn't have the most up to date gh-pages branch on which to rebase.

Fixes #60 

Trying:
 - Fetch GH Pages before mike --rebase

### CODE REVIEW

**Comment by e-lo** [comment]
@botanize I'm not sure why pushing this didn't trigger a GH action to test it?  🤔

**Comment by botanize** [comment]
Looks like the push target only targets pushes to the TIDES repo in the TIDES-transit org. So pushes to your fork aren't being run. We'll need to add a pull-request target to the workflow to run on pull-requests. I'm ok approving this as is an seeing if it works. But feel free to add a pull-request target to the build docs workflow.

**Comment by e-lo** [comment]
@botanize  - I'm not sure if we can see the effects of this until it is merged to `main`?

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:13]: I like the workflow_dispatch trigger, didn't know about that one! I'm not sure about create though. Do we really want to run it for every branch created?
I think the paths restrictions are too strict, they'd need to also include the configuration files, docs/document.py, and the contents of spec/, and now that's pretty much all of the files!

Maybe just 

```yaml
on:
  push:
  pull_request:
  workflow_dispatch:
```

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:45]: delete this line

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:47]: nice

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:3]: remove trailing whitespace

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:60]: add newline at end of file

**Review by botanize — CHANGES_REQUESTED** [review]
see comments

File: .github/workflows/docs.yml
Comment (e-lo) [reply .github/workflows/docs.yml]: I dropped the path search but kept 'create' b/c I think we want it to create for branches and tags for now...we can drop if/when it starts being too much.

Note that the delete function will delete the docs for any reference (branch/tag) that we delete from the repo so we shouldn't get too crowded.

**Review by botanize — APPROVED** [review]

**Review by botanize — CHANGES_REQUESTED** [review]
There were two workflow failures for this PR, can you take a look at those and see if something needs to change?

**Comment by e-lo** [comment]
Failure Report:

Mike doesn't like versions with non slug-a-ble names (i.e. `merge/62`)

- [ ] Turn GH refs into slugs

GH Action `actions/github-script@v5` got a deprecation warning and other icky stuff that I don't think I can fix quickly (and unsure of ROI). 

- [ ] Delete the triggering job from `workflow.yml`
- [ ] Document as an outstanding issue #69

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:20]: I'm seeing an error in the [workflow](https://github.com/TIDES-transit/TIDES/actions/runs/3153704737) that a step can't have both `uses` and `run` keys

File: .github/workflows/docs.yml
Comment (botanize) [inline .github/workflows/docs.yml:41]: missing new line

**Review by botanize — CHANGES_REQUESTED** [review]

**Comment by botanize** [comment]
I was just able to manually update the documentation by first running the "clean", then the "build-deploy" workflows. So maybe this is more of an ordering issue?

**Comment by e-lo** [comment]
@SorenSpicknall If you have time to troubleshoot outstanding issues in this PR it would be great.

**Pull request merged**
